osbenchmark/builder/executors/executor.py

```diff
@@ -1,17 +1,31 @@
-"""
-Executors are used to run shell commands on the cluster hosts. Implementations of this class will use various
-technologies to interface with the hosts of a cluster.
-"""
+import logging
+
+
 class Executor:
+    """
+    Executors are used to run shell commands on the cluster hosts. Implementations of this class will use various
+    technologies to interface with the hosts of a cluster.
+    """
     def __init__(self):
-        pass
+        self.logger = logging.getLogger(__name__)
 
-    def execute(self, host, commands):
+    def execute(self, host, command, **kwargs):
         """
         Executes a list of commands against the provided host
 
         ;param host: A Host object defining the host on which to execute the commands
-        ;param commands: A list of strings defining shell commands
-        ;return return_code: The return code of the execution
+        ;param command: A shell command as a string
+        ;return output: The output of the command
+        """
+        raise NotImplementedError
+
+    def copy(self, host, source, destination):
```

Comment:
`copy` is added to deal with the future challenge of moving files to external hosts. The SSMExecutor for self-managed EC2 clusters will need to setup a [session](https://www.tripwire.com/state-of-security/security-data-protection/cloud/aws-session-manager-enhanced-ssh-scp-capability/) to copy files to EC2 hosts rather than simply using `run-command` with shell commands.

> perhaps there's would be other scenarios for which we should consider having methods here too?

This is very possible. I expect a few slight nuances like this to appear as I work through decoupling each builder "action" from its "location".